test/Microsoft.AspNetCore.Mvc.Localization.Test/HtmlLocalizerTest.cs

```diff
@@ -158,6 +159,32 @@ public void HtmlLocalizer_HtmlWithInvalidResourceString_ContentThrowsException(s
             Assert.Equal("Input string was not in a correct format.", exception.Message);
         }
 
+        /// <summary>
+        /// This tests that LocalizedString instances containing curly braces (e.g. JSON or code snippets) can be used without a FormatException when no arguments are provided.
+        /// </summary>
+        [Fact]
+        public void HtmlLocalizer_HtmlWithJsonText_ReturnsLocalizedHtmlString()
+        {
+            // Arrange
+            var localizedString = new LocalizedString("ExampleJson", "example JSON: {\"foo\":\"bar\"}");
```

Comment:
Could you look at the tests below? They use `TestStringLocalizer` which has a convenience` htmlLocalizer.GetString` method that might do what you're currently doing.